k8s/src/main/scala/io/buoyant/k8s/Ns.scala

```diff
@@ -0,0 +1,93 @@
+package io.buoyant.k8s
+
+import com.twitter.conversions.time._
+import com.twitter.finagle.service.Backoff
+import com.twitter.finagle.tracing.Trace
+import com.twitter.finagle.util.DefaultTimer
+import com.twitter.util._
+
+abstract class Ns[O <: KubeObject: Manifest, W <: Watch[O]: Manifest, L <: KubeList[O]: Manifest, Cache](
+  backoff: Stream[Duration] = Backoff.exponentialJittered(10.milliseconds, 10.seconds),
+  timer: Timer = DefaultTimer.twitter
+) {
+  // note that caches must be updated with synchronized
+  private[this] val caches = Var[Map[String, Cache]](Map.empty[String, Cache])
+  // XXX once a namespace is watched, it is watched forever.
+  private[this] var _watches = Map.empty[String, Activity[Closable]]
+  def watches = _watches
```

Comment:
nit: would be nice to annotate the type since this is basically a public interface now